src/models/kelvinvoigt.jl

```diff
@@ -53,14 +52,13 @@ FractS_KelvinVoigt =  RheoModelClass(
         J = quote
               (t^(a)/cₐ)*mittleff(a, 1 + a, -k*t^a/cₐ)
             end,
-        Ja_safe = false,
         # Storage modulus
         Gp = quote
-                if a!=1.0
+                #if a!=1.0        #  This is to be removed as the condition contradicts the constraints
```

Comment:
This one:
```
# Constraints
        constraint = quote
                 (a<1) & (a>0)
                end,
```

For the floating-point error, maybe it would make sense to use the specialised functions that solve this problem?
```
julia> cos(pi/2)
6.123233995736766e-17

julia> cospi(1/2)
0.0
```

Reply:
That constraint is only enforced in a handful of places though (e.g. checking initial conditions before the fit process starts). Importantly, it's not enforced when making a direct call to the modulus (I don't think, unless you/Ale changed it), or during optimisation process. (Indeed, we wouldn't need the special mittleff wrapper if the constraint was enforced throughout optimisation.)

Same applies to the other changes you made in a number of the the other models. (I didn't comment on them all, but I noticed there were several other changes you made getting rid of edge case handling. 

I do not have skin in the game here because I'm unlikely to use RHEOS in future, but what I can say for certain is that those edge cases were in their specifically because we hit them at various times using RHEOS in practice, and they caused unwanted behaviour - so we need to think carefully about each removal. Probably worth discussing with Ale also.

Reply:
That is a good point. It is not very important here, but this touches on a broader set of problems about how we check the constraints when we call the moduli. I'd prefer to keep it clean and fast for the low level access like _G and _Ga, but we should definitely make sure that the parameters sent to functions such as relaxmod are checked properly, for instance for the relative values of the exponents, etc. There are at the moment many ways it could go wrong, and not all of them can be hacked as we do here - we need to throw exceptions I think. There are indeed other issues with the fact that NLopt does not enforce the constraints beyond checking parameters at the start. Maybe this deserves its own issue...

Reply:
I did some work in that direction already, which is documented in the open issue https://github.com/JuliaRheology/RHEOS.jl/issues/86

Exceptions make sense in some contexts, but definitely not all of them (the above code with cos(pi/2) being a good example). A 'smooth' transition from springpot to spring/dashpot specialisation causes fundamental difficulties in computational implementation, but the conversation around to what extent we want to support it and how is still an important one.

Reply:
Yes, not all contexts can be addressed like that indeed.